packages/device_info_plus/lib/device_info_plus_tizen.dart

```diff
@@ -0,0 +1,155 @@
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:device_info_plus_platform_interface/device_info_plus_platform_interface.dart';
+import 'package:flutter/services.dart';
+
+/// Information derived from `system_info`.
+///
+/// See: https://docs.tizen.org/application/native/guides/device/system
+class TizenDeviceInfo {
+  /// Tizen device info class.
+  TizenDeviceInfo({
+    required this.modelName,
+    required this.cpuArch,
+    required this.nativeApiVersion,
+    required this.platformVersion,
+    required this.webApiVersion,
+    required this.buildDate,
+    required this.buildId,
+    required this.buildString,
+    required this.buildTime,
+    required this.buildType,
+    required this.buildVariant,
+    required this.buildRelease,
+    required this.deviceType,
+    required this.manufacturer,
+  });
+
+  /// http://tizen.org/system/model_name
+  ///
+  /// The value is an empty String if it is not available.
+  final String modelName;
+
+  /// http://tizen.org/feature/platform.core.cpu.arch
+  ///
+  /// The value is an empty String if it is not available.
+  final String cpuArch;
+
+  /// http://tizen.org/feature/platform.native.api.version
+  ///
+  /// The value is an empty String if it is not available.
+  final String nativeApiVersion;
+
+  /// http://tizen.org/feature/platform.version
+  ///
+  /// The value is an empty String if it is not available.
+  final String platformVersion;
+
+  /// http://tizen.org/feature/platform.web.api.version
+  ///
+  /// The value is an empty String if it is not available.
+  final String webApiVersion;
+
+  /// http://tizen.org/system/build.date
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildDate;
+
+  /// http://tizen.org/system/build.id
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildId;
+
+  /// http://tizen.org/system/build.string
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildString;
+
+  /// http://tizen.org/system/build.time
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildTime;
+
+  /// http://tizen.org/system/build.type
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildType;
+
+  /// http://tizen.org/system/build.variant
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildVariant;
+
+  /// http://tizen.org/system/build.release
+  ///
+  /// The value is an empty String if it is not available.
+  final String buildRelease;
+
+  /// http://tizen.org/system/device_type
+  ///
+  /// The value is an empty String if it is not available.
+  final String deviceType;
+
+  /// http://tizen.org/system/manufacturer
+  ///
+  /// The value is an empty String if it is not available.
+  final String manufacturer;
+
+  /// Deserializes from the message received from [_kChannel].
+  static TizenDeviceInfo fromMap(Map<String, dynamic> map) {
```

Comment:
@HakkyuKim I'm just curious and asking. I am a dart newbie. :) why do you use static method here instead of factory constructor?

Reply:
It's not always the case but generally speaking, the factory constructor should be used only when it's necessary. (e.g. The constructor might not create a new instance and instead return an instance from a cache, or it might return an instance of a subtype.)

Reply:
@WonyoungChoi
I'm glad to ponder about this intersting question.
Honestly, I haven't given much thought into the current code style, since the focus of this PR was to change only the necessary code to work with the `device_info_plus` package. 

`fromMap` reminded me of a similar method named `fromJson` from [Json serializing documentation](https://flutter.dev/docs/development/data-and-backend/json#serializing-json-inside-model-classes).
At the time, I didn't notice that `fromJson` was a named constructor because a static method that returns a new instance and a named constructor both seemed appropriate in simple data type transformation(`Map` to `DeviceInfoTizen`).

However, if the method was a factory constructor, to me at least, would have given me the impression that some untrivial actions may be performed additionally before returning an instance.
Nevertheless, a named constructor seems more appropriate, as done with `fromJson`, for consistency sake and becuase it always guarentees returning an instance.

@swift-kim also made a good point about factory constructors canonically being used to return an instance of a subtype.

Reply:
All other platform implementations of `XXXDeviceInfo` use static method `fromMap`, `TizenDeviceInfo` should follow this for consistency. See https://github.com/fluttercommunity/plus_plugins/blob/a50f5ada496155d71c98040043b642bfdaa48b49/packages/device_info_plus_platform_interface/lib/model/android_device_info.dart#L114-L139 for Android.

Reply:
Using the factory keyword and implementing the factory pattern with a static method are functionally the same. In the end it doesn't matter. 😅 I found the related article about this: https://dash-overflow.net/articles/factory/. 
@HakkyuKim  Thanks for your kind replies to my questions.